CMakeLists.txt

```diff
@@ -8,6 +8,7 @@ if(POLICY CMP0074)
 endif()
 
 # Set target C++ standard and required compiler features
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
Can we move this to a line 31/32, after the project is set. Also add a comment indicating what forced us to turn it on.